web/src/routes/admin/system-settings/+page.svelte

```diff
@@ -58,6 +60,25 @@
     setTimeout(() => downloadManager.clear(downloadKey), 5000);
   };
 
+  const uploadConfig = () => {
+    const fileSelector = document.createElement('input');
+    fileSelector.type = 'file';
+    fileSelector.accept = '.json';
+    fileSelector.addEventListener('change', (e) => {
+      const file = (e.target as HTMLInputElement).files?.[0];
+      if (!file) {
+        return;
+      }
```

Comment:
I think this should just be a normal input element in the html stuff below. It can be hidden if you want though. But button click just needs to trigger clicking the input. It doesn't need to be dynamically created. You can use something like this:

```ts
let inputElement: HTMLInputElement;
const handleUpload = () => inputElement?.click();
..

<input bind:this={inputElement} ... />

Reply:
I based myself on [this](https://github.com/immich-app/immich/blob/a2180a467d09c01ec41c1481e51c7485d5a82046/web/src/lib/utils/file-uploader.ts#L26) but I'm happy to make that change if you wish.

Reply:
Ah but that is a pure TypeScript file without HTML, right? Unlike this page. Sorry, learning as I go!